### ISSUE TITLE
SelectiveAdam decides dense-vs-rowwise from a magic string

### ISSUE THREAD

**Issue by nandometzger** [nandometzger/metal-gauss#11]
`SelectiveAdam.add_param_group` in `metal_gauss/selective_adam.py`:

```python
group.setdefault("rowwise", group.get("name") != "appearance")
```

Correct today — `train.py` does pass `"name": "appearance"` when the appearance model is enabled. But the behaviour of an optimizer group is decided by comparing a string to a literal, so the next non-Gaussian group added without that exact name is treated rowwise. It will then either raise

```
rowwise SelectiveAdam parameters must have at least one row per visibility-mask entry
```

or, if its first dimension happens to be large enough, silently take per-row updates driven by a Gaussian visibility mask that has nothing to do with it.

## The change

Pass `"rowwise": False` explicitly at the call site in `train.py` and drop the name comparison from `add_param_group`. Keep the default `True`, so existing Gaussian groups are unaffected.

`tests/test_trainer.py` already covers the appearance group taking ordinary Adam and matching `torch.optim.Adam` — that test should keep passing unchanged, which is the point.

Came out of reviewing #2.
